packages/transaction_signing_gateway/lib/wallet_derivator.dart

```diff
@@ -0,0 +1,12 @@
+import 'package:dartz/dartz.dart';
+import 'package:transaction_signing_gateway/model/network_info.dart';
+import 'package:transaction_signing_gateway/model/private_wallet_credentials.dart';
+import 'package:transaction_signing_gateway/model/wallet_derivation_failure.dart';
+
+abstract class WalletDerivator {
+  Either<WalletDerivationFailure, PrivateWalletCredentials> derive({
+    required String mnemonic,
+    required String walletAlias,
+    required NetworkInfo networkInfo,
+  });
```

Comment:
we cannot accept NetworkInfo here, as this model is specific to Alan and Cosmos network. what we can do instead is to accept following:

```suggestion
  Either<WalletDerivationFailure, PrivateWalletCredentials> derive({
    required WalletDerivationInfo walletDerivationInfo
  });
```
Add new interface for `WalletDerivationInfo`:

```dart
abstract class WalletDerivationInfo {
}
```

and implementation specific for Alan

```dart
class AlanWalletDerivationInfo implements WalletDerivationInfo {
    final String mnemonic;
    final String walletAlias;
    final NetworkInfo networkInfo; // Alan's version of network info, we don't need a separate model in the transaction_signing_gateway library
    
}
```

Reply:
✅